src/main/kotlin/biokotlin/genome/MAFProcessingUtils.kt

```diff
@@ -447,4 +452,321 @@ fun getCoverageIdentityPercentForMAF(mafFile:String, region:String = "all"): Dat
+        //chromosome names are not the same -> no overlap
+        mafRecord1.refRecord.chromName != mafRecord2.refRecord.chromName -> false
+        //record1 end comes before record2 -> no overlap
+        mafRecord1.refRecord.start + mafRecord1.refRecord.size <= mafRecord2.refRecord.start -> false
+        //record1 start comes after record2 -> no overlap
+        mafRecord1.refRecord.start >= mafRecord2.refRecord.start + mafRecord2.refRecord.size -> false
+        //otherwise -> overlap
+        else -> true
+    }
+}
+
+/**
+ * [target] is a mutable list of maf records that are assumed to be sorted by chromosome and start. If there are
+ * reference positions present in source that are absent from target then the sequence for those positions
+ * will be added to target.
+ * MAF start positions are 0-based numbers
+ */
+fun augmentList(target: MutableList<MAFRecord>, source: List<MAFRecord>)  {
+    /**
+    target assumed to be sorted.
+    This method will add blocks or parts of blocks from source not covered in target.
+    It follows these steps:
+    1.  Sorts the records by chromosome
+    2. Create a RangeMap from source list
+    3.  For each cheomosome
+    a. For each gap in target list
+    1. make a range of the entire gap
+    2. find the blocks from source that overlap the gap range
+    3. extract maf records from those blocks that fall within the gap, this will require block splitting
+    4. add those records to target
+     **/
+
+    //create a map of chromosome name -> RangeMap
+    val chromToRangesSRC = mutableMapOf<String, ArrayList<MAFRecord>>()
+    val chromToRanges = mutableMapOf<String, RangeMap<Int, MAFRecord>>() // needed for pulling sequence from submap
+    for (mafrec in source) {
+        val chrom = mafrec.refRecord.chromName
+        // this needed for creating the list of gap ranges
+        var chrMAFRecList = chromToRangesSRC[chrom]
+        if (chrMAFRecList == null) {
+            chrMAFRecList = ArrayList<MAFRecord>()
+            chromToRangesSRC[chrom] =  chrMAFRecList
+        }
+        chrMAFRecList!!.add( mafrec)
+
+        // create a map of chromosome name -> RangeMap
+        // This one needed for getting subranges later from the gaps
+        var chrMap = chromToRanges[chrom]
+        if (chrMap == null) {
+            chrMap = TreeRangeMap.create()
+            chromToRanges[chrom] = chrMap
+        }
+        val start = mafrec.refRecord.start
+        val end = mafrec.refRecord.start + mafrec.refRecord.size - 1
+        chrMap!!.put(Range.closed(start,end), mafrec)
+    }
+
+    // GEt the target positions
+    // This creates a map of chromName (key) to a list of MAFtoGVCFPlugin.MAFRecords (value)
+    val chromToRangesTGT = target.groupBy{it.refRecord.chromName}
+
+    val addedBlocks = mutableListOf <MAFRecord>()
+    // Not all chromosomes will be in both target and src
+    println("augmentList: number of chromosomes in chromToMAFRecords = ${chromToRangesSRC.keys.size}")
```

Comment:
Use the logger instead of prinlns

Reply:
Biokotlin doesn't have a logging system setup.  Jeffrey added a logger file to 1 of his featureTree classes, but it doesn't name the class as is done with PHG.  Throughout BioKotlin we have println instead of logger.  I will write a trello card to deal with BioKotlin logging - we need to do something consistent and repository wide